controller/rest/dlp_rule.go

```diff
@@ -499,10 +499,16 @@ func createDlpSensor(w http.ResponseWriter, conf *api.RESTDlpSensorConfig, cfgTy
 		sensor.RuleListNames[rdr.Name] = rdr.Name
 	}
 	//save full rule with pattern in default sensor
-	clusHelper.PutDlpSensor(defsensor, false)
+	if err := clusHelper.PutDlpSensor(defsensor, false); err != nil {
+		log.WithFields(log.Fields{"error": err}).Error("PutDlpSensor")
+		return err
+	}
 
 	//create new sensor
-	clusHelper.PutDlpSensor(sensor, true)
+	if err := clusHelper.PutDlpSensor(sensor, true); err != nil {
+		log.WithFields(log.Fields{"error": err}).Error("PutDlpSensor")
+		return err
```

Comment:
No restRespErrorMessage()-related function is called before returning error.
See createWafSensor() for reference.